src/lib/abstractimagedownloader.cpp

```diff
@@ -98,8 +92,9 @@ void AbstractImageDownloaderPrivate::manageStack()
     }
 }
 
-static bool writeImageData(const QString &localFilePath, QNetworkReply *reply)
+bool AbstractImageDownloaderPrivate::writeImageData(ImageInfo *info, QNetworkReply *reply, QString *outFileName)
```

Comment:
Is there some reason why outFileName is a pointer instead of a reference?
(Code is not prepared to handle null value etc)

Reply:
Encountering foo(&bar) one can immediately tell that bar might get some changes. On foo(bar) the default assumption is no changes to bar.